spec/aptible/cli/subcommands/operation_spec.rb

```diff
@@ -26,4 +28,67 @@
       subject.send('operation:cancel', 1)
     end
   end
+  describe '#operation:logs' do
+    it 'sends operation logs request when subcommand sent successfully' do
+      operation_id = SecureRandom.uuid
+      expect(Aptible::Api::Operation).to receive(:find).with(1, token: token)
+        .and_return(Fabricate(
+                      :operation, status: 'succeeded', id: operation_id
+        ))
+
+      # stub out operations call
+      response = Net::HTTPSuccess.new(1.0, '301', 'OK')
+      response.add_field(:location, 'https://s3.aptible.com/not-real/s3')
+
+      # stub out s3 call
+      s3_response = instance_double(Net::HTTPResponse, body: 'Mock logs')
+
+      allow(Net::HTTP).to receive(:new).twice do |_, _, _|
+        net_http_double
+      end
+      expect(net_http_double).to receive(:use_ssl=).twice
+      expect(net_http_double).to receive(:request).twice do |request|
+        if request.path == "/operations/#{operation_id}/logs"
+          response
+        elsif request.path == '/not-real/s3'
+          s3_response
+        end
+      end
+
+      subject.send('operation:logs', 1)
+    end
+    it 'errors when operation is not found' do
+      expect(Aptible::Api::Operation).to receive(:find).with(1, token: token)
+        .and_return(nil)
+
+      expect { subject.send('operation:logs', 1) }
+        .to raise_error('Operation #1 not found')
+    end
+    it 'errors when operation is not status expected' do
+      operation_id = SecureRandom.uuid
+      expect(Aptible::Api::Operation).to receive(:find).with(1, token: token)
+        .and_return(Fabricate(:operation, status: 'queued', id: operation_id))
+
+      expect { subject.send('operation:logs', 1) }
+        .to raise_error('Error - You can view the logs when operation '\
+                        'is complete.')
+    end
+    it 'errors when operation not found and errored on deploy API' do
```

Comment:
This test description confused me.

```suggestion
    it 'errors when operation logs are not found' do
```

Reply:
i can do that, i meant to say it failed on deploy api's return value, but your update is more ambiguous and still relevant